vng_api_common/client.py

```diff
@@ -1,17 +1,67 @@
 """
-Interface to get a zds_client object for a given URL.
+Interface to get a client object for a given URL.
 """
 
-from typing import Optional
+import logging
+import time
+from typing import Any, Optional
 
-from django.apps import apps
 from django.conf import settings
 from django.utils.module_loading import import_string
 
-from zds_client import Client
+import jwt
+from ape_pie import APIClient
+from requests import JSONDecodeError, RequestException, Response
 
+JWT_ALG = "HS256"
 
-def get_client(url: str, url_is_api_root=False) -> Optional[Client]:
+
+logger = logging.getLogger(__name__)
+
+
+class ClientError(RuntimeError):
+    pass
+
+
+# TODO: use more approriate method name
+def to_internal_data(response: Response) -> dict | list | None:
+    try:
+        response_json = response.json()
+    except JSONDecodeError:
+        logger.exception("Unable to parse json from response")
+        response_json = None
+
+    try:
+        response.raise_for_status()
+    except RequestException as exc:
+        if response.status_code >= 500:
+            raise
+        raise ClientError(response_json if response_json is not None else {}) from exc
```

Comment:
We could but we might lose some interesting information whenever requests are not done correctly (e.g `400` error status codes from responses). This could be used for logging purposes like it was done before, see [7fa83c5dc0e294fe430c819f9993532cc878293f](https://github.com/maykinmedia/gemma-zds-client/commit/7fa83c5dc0e294fe430c819f9993532cc878293f) (if an application does not want to use django-log-outgoing-requests and use its own).